src/VisualStudio/Core/Def/ValueTracking/ValueTrackingToolWindow.cs

```diff
@@ -49,28 +64,33 @@ public ValueTrackingToolWindow() : base(null)
             Content = _root;
         }
 
-        public ValueTrackingToolWindow(ValueTrackingTreeViewModel viewModel)
-            : base(null)
+        public void Initialize(ValueTrackingTreeViewModel viewModel, Workspace workspace, IThreadingContext threadingContext)
         {
-            Caption = ServicesVSResources.Value_Tracking;
-            Content = _root;
+            Contract.ThrowIfTrue(Initialized);
+
+            Initialized = true;
             ViewModel = viewModel;
+            _workspace = workspace;
+            _threadingContext = threadingContext;
+
+            _workspace.WorkspaceChanged += OnWorkspaceChanged;
         }
 
-        public TreeItemViewModel? Root
+        private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
         {
-            get => ViewModel?.Roots.Single();
-            set
-            {
-                if (value is null)
-                {
-                    return;
-                }
+            Contract.ThrowIfFalse(Initialized);
 
-                Contract.ThrowIfNull(ViewModel);
+            if (e.Kind is WorkspaceChangeKind.SolutionCleared
+                or WorkspaceChangeKind.SolutionRemoved)
```

Comment:
```suggestion
            if (e.Kind is WorkspaceChangeKind.SolutionCleared
                       or WorkspaceChangeKind.SolutionRemoved)
```

Reply:
![GreysAnatomyAtticusLincolnGIF](https://user-images.githubusercontent.com/475144/179098291-0ed64163-83cd-4286-9d92-51bcb0b6deb6.gif)

Reply:
@ryzngard But then it's all nice and aligned!

Reply:
I prefer +1 indentation over alignment 🤷, but not enough to argue over it. Only enough to grumble while I change it